Dev Environment Breaks with MySQL 8.3 During Installation

## Description
When setting up the development environment with MySQL version 8.3, `pip install -e ".[dev]"`  fails. 

## Expected Behavior 
Running `pip install -e ".[dev]"` should successfully install all necessary development dependencies without errors, even with MySQL 8.3.

## Current Behavior
The installation process breaks when attempting `pip install -e ".[dev]"`. The following errors are encountered:
```
src/MySQLdb/_mysql.c:527:9: error: call to undeclared function 'mysql_ssl_set'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
              mysql_ssl_set(&(self->connection), key, cert, ca, capath, cipher);
              ^
      src/MySQLdb/_mysql.c:527:9: note: did you mean 'mysql_close'?
      /usr/local/Cellar/mysql/8.3.0/include/mysql/mysql.h:797:14: note: 'mysql_close' declared here
      void STDCALL mysql_close(MYSQL *sock);
                   ^
      src/MySQLdb/_mysql.c:1795:9: error: call to undeclared function 'mysql_kill'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
          r = mysql_kill(&(self->connection), pid);
              ^
      src/MySQLdb/_mysql.c:1795:9: note: did you mean 'mysql_ping'?
      /usr/local/Cellar/mysql/8.3.0/include/mysql/mysql.h:525:13: note: 'mysql_ping' declared here
      int STDCALL mysql_ping(MYSQL *mysql);
                  ^
      src/MySQLdb/_mysql.c:2011:9: error: call to undeclared function 'mysql_shutdown'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
          r = mysql_shutdown(&(self->connection), SHUTDOWN_DEFAULT);
              ^
      3 errors generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for mysqlclient
Successfully built feast
Failed to build mysqlclient
ERROR: Could not build wheels for mysqlclient, which is required to install pyproject.toml-based projects
```

## Steps to reproduce
1. Create and activate a new Conda environment for `feast`:
 ```
 conda create --name feast python=3.9
 conda activate feast
 ```
 2. Install dependencies:
   ```
   pip install pip-tools
   brew install mysql@8.3
   brew install xz protobuf openssl zlib
   pip install cryptography -U
   conda install protobuf
   conda install pymssql
   pip install -e ".[dev]"
   ```

--- a/sdk/python/feast/infra/feature_servers/multicloud/Dockerfile
+++ b/sdk/python/feast/infra/feature_servers/multicloud/Dockerfile
@@ -4,7 +4,6 @@ RUN apt update && \
         apt install -y \
         jq \
         python3-dev \
-        default-libmysqlclient-dev \
         build-essential
 
 RUN pip install pip --upgrade

--- a/sdk/python/feast/infra/feature_servers/multicloud/Dockerfile.dev
+++ b/sdk/python/feast/infra/feature_servers/multicloud/Dockerfile.dev
@@ -4,7 +4,6 @@ RUN apt update && \
         apt install -y \
         jq \
         python3-dev \
-        default-libmysqlclient-dev \
         build-essential
 
 RUN pip install pip --upgrade

--- a/sdk/python/requirements/py3.10-ci-requirements.txt
+++ b/sdk/python/requirements/py3.10-ci-requirements.txt
@@ -472,8 +472,6 @@ mypy-extensions==1.0.0
     #   mypy
 mypy-protobuf==3.1.0
     # via feast (setup.py)
-mysqlclient==2.2.0
-    # via feast (setup.py)
 nbclient==0.9.0
     # via nbconvert
 nbconvert==7.11.0

--- a/sdk/python/requirements/py3.8-ci-requirements.txt
+++ b/sdk/python/requirements/py3.8-ci-requirements.txt
@@ -488,8 +488,6 @@ mypy-extensions==1.0.0
     #   mypy
 mypy-protobuf==3.1.0
     # via feast (setup.py)
-mysqlclient==2.2.0
-    # via feast (setup.py)
 nbclient==0.9.0
     # via nbconvert
 nbconvert==7.11.0

--- a/sdk/python/requirements/py3.9-ci-requirements.txt
+++ b/sdk/python/requirements/py3.9-ci-requirements.txt
@@ -479,8 +479,6 @@ mypy-extensions==1.0.0
     #   mypy
 mypy-protobuf==3.1.0
     # via feast (setup.py)
-mysqlclient==2.2.0
-    # via feast (setup.py)
 nbclient==0.9.0
     # via nbconvert
 nbconvert==7.11.0

--- a/sdk/python/tests/unit/test_sql_registry.py
+++ b/sdk/python/tests/unit/test_sql_registry.py
@@ -103,7 +103,7 @@ def mysql_registry():
 
     registry_config = RegistryConfig(
         registry_type="sql",
-        path=f"mysql+mysqldb://{POSTGRES_USER}:{POSTGRES_PASSWORD}@127.0.0.1:{container_port}/{POSTGRES_DB}",
+        path=f"mysql+pymysql://{POSTGRES_USER}:{POSTGRES_PASSWORD}@127.0.0.1:{container_port}/{POSTGRES_DB}",
     )
 
     yield SqlRegistry(registry_config, "project", None)

--- a/setup.py
+++ b/setup.py
@@ -116,7 +116,7 @@ POSTGRES_REQUIRED = [
     "psycopg2-binary>=2.8.3,<3",
 ]
 
-MYSQL_REQUIRED = ["mysqlclient", "pymysql", "types-PyMySQL"]
+MYSQL_REQUIRED = ["pymysql", "types-PyMySQL"]
 
 HBASE_REQUIRED = [
     "happybase>=1.2.0,<3",

